src/js/libcs/Render2D.js

```diff
@@ -619,6 +619,36 @@ Render2D.drawline = function(homog) {
     }
 };
 
+// draws a segment consisting of 2 rays
+Render2D.drawRaySegment = function(A, B, C, sflip) {
+    var ptA = eval_helper.extractPoint(A);
+    var ptB = eval_helper.extractPoint(B);
+    var ptC = eval_helper.extractPoint(C);
+    if (!ptA.ok || !ptB.ok || !ptC.ok) {
+        return;
+    }
+
+    var dx = sflip * (ptA.x - ptB.x);
+    var dy = sflip * (ptA.y - ptB.y);
+    var norm = Math.sqrt(dx * dx + dy * dy);
+
+    // get points outside canvas (at "infinity")
+    var sc = csport.drawingstate.matrix.sdet;
+    var farAway = 25000 / sc; // 25000px in user coordinates
```

Comment:
This is less than ideal, but better than what we have now. We should keep in mind that it would be nice to have a better solution for this one day. Particularly for infinite line segments whose endpoints happen to be more than 25000px outside the current viewport area, so that we don't see a ray that should be visible in theory.